### ISSUE TITLE
Do we need a map loss for the decoder? 

### ISSUE THREAD

**Issue by AidanRocke** [AidanRocke/variational_empowerment#6]
1. I wonder whether there's any empirical evidence that a loss of this type might help:
    
```
    def map_decoder_loss(self):
        
        return tf.square((self.source_action-self.decoder_mu)/tf.exp(self.decoder_log_sigma))
```

so in the training script we would have: 

`decoder_loss = tf.reduce_mean(tf.multiply(tf.constant(-1.0),log_decoder)) + tf.reduce_mean(map_loss)
`

2. We already have bounds for the per-action source and decoder distribution means and variances so the question is whether a MAP loss would really help. 

**Note:**  Here's a Wikipedia reference on MAP losses.

**Comment by AidanRocke** [comment]
Upon further reflection, I don't see any reason why this would be a good idea. The decoder is an inverse model which involves determining what to do(i.e. the action) in order to go from state s to s'. 
The source on the other hand is used for exploration. 

They have two very different purposes.
